lib/takeaway.rb

```diff
@@ -0,0 +1,65 @@
+require 'twilio-ruby'
+
+class Takeaway
+
+  attr_accessor :menu, :dishes_selected, :total_price, :total_price_array, :dishes_available, :dish_price
+
+  def initialize
+    @menu = Hash.new
+    @dishes_selected = []
+  end
+
+  def menu_update(dish)
+    menu["#{dish.name}"] = dish.price.to_s + "£"
+  end 
+
+  def order
+    puts self.menu
+    puts "To begin the selection, press enter. To cancel the action, type 'exit'."
+    @total_price_array = []
+
+    while gets.chomp != 'exit' 
+      puts "Please choose a dish."
+      dish = gets.chomp
+      dishes_selected << dish
+
+      puts "Please select the quantity of the dish."
+      quantity = gets.chomp
+
+      puts "Please press enter to select another dish or type 'exit'."
+
+      dish_price_calculator(dish, quantity)
+    end
+
+    puts "Here is your order."
+    puts dishes_selected
+    total_price_calculator
+  end
+
+  def dish_price_calculator(dish, quantity)
+    price = menu["#{dish}"]
+    dish_price = price.to_i * quantity.to_i 
+    total_price_array << dish_price
+  end
+
+  def total_price_calculator
+    @total_price = total_price_array.inject {|result, i| result + i}
+    p total_price.to_s  + '£'
+    raise 'Total price is wrong' if total_price != total_price_array.inject {|result, i| result + i}
+  end
+
+  # 'send_reply_message' method not implemented, it should have been part of the 'order' method
+
+  def send_reply_message
+  account_sid = ''
+  auth_token = ''
+  @client = Twilio::REST::Client.new account_sid, auth_token
+
+  @client.messages.create(
+  from: '',
+  to: '',
+  body: 'Your order has been accepted and is already on the way. Thanks for choosing Takeaway.rb.'
+)
+
+  end
+end
```

Comment:
Hey Tommaso,

I wouldn’t say that adding classes adds dependencies. 

Classes should be encapsulated so that they are not dependent on another class to function. That is one of the hardest skills to develop in OOP, but I am sure you will pick it up. 

I see you have sent a bunch of emails - maybe the best would be for us to have a 121 so that we can talk over your code?

Let me know when works for you.

Cheers,

___________________________

Dan Le Dosquet-Bergquist, Meta Maker

+44 753 900 8999 | www.makersacademy.com <http://www.makersacademy.com/>

GitHub <https://github.com/danldb>    |    Twitter <http://www.twitter.com/danldb>
> On 19 Feb 2015, at 14:25, Tommaso Bratto <notifications@github.com> wrote:
> 
> In lib/takeaway.rb <https://github.com/makersacademy/takeaway-challenge/pull/22#discussion_r24990534>:
> 
> > +
> > +  # 'send_reply_message' method not implemented, it should have been part of the 'order' method
> > +
> > +  def send_reply_message
> > +  account_sid = ''
> > +  auth_token = ''
> > +  @client = Twilio::REST::Client.new account_sid, auth_token
> > +
> > +  @client.messages.create(
> > +  from: '',
> > +  to: '',
> > +  body: 'Your order has been accepted and is already on the way. Thanks for choosing Takeaway.rb.'
> > +)
> > +
> > +  end
> > +end
> Ok, but wouldn't using more classes potentially create more dependencies? For example if I needed to make various objects interact, such as the customer selecting the dish and reporting its quantity to the takeaway, wouldn't I need to use more than one object as arguments of an hypothetical customer.order method? (like customer.order(dish, takeaway))
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/makersacademy/takeaway-challenge/pull/22/files#r24990534>.
>